src/main/java/io/github/mfvanek/pg/connection/HighAvailabilityPgConnectionImpl.java

```diff
@@ -53,12 +67,46 @@ public Set<PgConnection> getConnectionsToAllHostsInCluster() {
 
     @Nonnull
     public static HighAvailabilityPgConnection of(@Nonnull final PgConnection connectionToPrimary) {
-        return new HighAvailabilityPgConnectionImpl(connectionToPrimary, Collections.singleton(connectionToPrimary));
+        final PrimaryHostDeterminer primaryHostDeterminer = new PrimaryHostDeterminerImpl();
+        return new HighAvailabilityPgConnectionImpl(connectionToPrimary, Collections.singleton(connectionToPrimary), primaryHostDeterminer);
     }
 
     @Nonnull
     public static HighAvailabilityPgConnection of(@Nonnull final PgConnection connectionToPrimary,
                                                   @Nonnull final Collection<PgConnection> connectionsToAllHostsInCluster) {
-        return new HighAvailabilityPgConnectionImpl(connectionToPrimary, connectionsToAllHostsInCluster);
+        final PrimaryHostDeterminer primaryHostDeterminer = new PrimaryHostDeterminerImpl();
+        return new HighAvailabilityPgConnectionImpl(connectionToPrimary, connectionsToAllHostsInCluster, primaryHostDeterminer);
+    }
+
+    @Nonnull
+    public static HighAvailabilityPgConnection of(@Nonnull final PgConnection connectionToPrimary,
+                                                  @Nonnull final Collection<PgConnection> connectionsToAllHostsInCluster,
+                                                  @Nullable final Integer delaySeconds) {
+        final PrimaryHostDeterminer primaryHostDeterminer = new PrimaryHostDeterminerImpl();
+        final HighAvailabilityPgConnectionImpl highAvailabilityPgConnection = new HighAvailabilityPgConnectionImpl(connectionToPrimary, connectionsToAllHostsInCluster, primaryHostDeterminer);
+        if (highAvailabilityPgConnection.connectionsToAllHostsInCluster.size() > 1 && delaySeconds != null) {
```

Comment:
Move this logic `if (highAvailabilityPgConnection.connectionsToAllHostsInCluster.size() > 1` to `startPrimaryUpdater`